File: Language/C#/Book/tmp/LINQ实战/LinqInAction/Chapter06to08.CSharp/Common/Subject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LinqInAction.Chapter06to08.Common
{
  public partial class Subject
  {
    public Guid ObjectId = Guid.NewGuid();

    public static void UpdateSubject(Subject changingSubject)
    {
      LinqBooksDataContext context = new LinqBooksDataContext();

      Subject existingSubject = context.Subjects.Where(subject => subject.SubjectId == changingSubject.SubjectId).FirstOrDefault();
      existingSubject.Name = changingSubject.Name;
      existingSubject.Description = changingSubject.Description;
      context.SubmitChanges();
    }
  }
}

File: Language/C#/Book/tmp/LINQ实战/Chapter06to08.VB/Common/Subject.vb
Partial Public Class Subject
    Public ObjectId As Guid = Guid.NewGuid()

    Public Shared Sub UpdateSubject(ByVal changingSubject As Subject)
        Dim context As New LinqBooksDataContext()
        Dim existingSubject As Subject = context.Subjects.Where(Function(s) s.SubjectId = changingSubject.SubjectId).FirstOrDefault()
        existingSubject.Name = changingSubject.Name
        existingSubject.Description = changingSubject.Description
        context.SubmitChanges()
    End Sub

End Class
